### PR TITLE
issue-46 - support dependency-only packages.

### PR DESCRIPTION
**Pull request by earlye**

### CODE REVIEW

**Comment by earlye** [comment]
See https://github.com/earlye/kyrios/pull/48 for an example of this in use. packages/stdbuf.yaml, in particular

File: src/kyrios.py [hunk 1]
```diff
@@ -85,12 +86,15 @@ def installPackage(packageName, package, context):
...
-    context['installedPackages'].append(packageName)
+    if 'packageManager' in platformConfig:
+        packageManagerName = platformConfig['packageManager']
+        packageManager = packageManagers[packageManagerName]
+        logging.debug("found packageManager: '{}': {}".format(packageManagerName, packageManager))
+
+        packageManager.installPackage(packageName, package, context, platformConfig)
+        context['installedPackages'].append(packageName)
+    elif not platformConfig.get('installedByDependency', False):
+        logging.warn("Package '{}' is required, but does not specify a `packageManager`, and is not `installedByDependency`.".format(packageName));
```
Comment (BlackVegetable) [inline src/kyrios.py:97]: Are you on a version that supports f-strings? If so, use those?

**Review by BlackVegetable — APPROVED** [review]
LGTM overall.

File: src/kyrios.py (hunk 1 above)
Comment (stevenuray) [reply src/kyrios.py:97]: If I remember correctly 2.7 is intentionally used here to be compatible with the default version of python found on many operating systems.

**Review by stevenuray — APPROVED** [review]

**Pull request merged**